src/plugins/intel_cpu/src/nodes/generic.cpp

```diff
@@ -17,8 +17,34 @@ namespace ov {
 namespace intel_cpu {
 namespace node {
 
+namespace {
+/**
+ * Dummy Shape Inference while Generic op doesn't support Dynamism
+ *
+ */
+class GenericShapeInfer : public ShapeInferEmptyPads {
+public:
+    GenericShapeInfer() = default;
+    std::vector<VectorDims> infer(
+        const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+        const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
+        IE_THROW(Unexpected) << "Generic operations doesn't support shape inference.";
+        return {};
+    }
+
+    port_mask_t get_port_mask() const override { return EMPTY_PORT_MASK; }
+};
+
+class GenericShapeInferFactory : public ShapeInferFactory {
+public:
+    ShapeInferPtr makeShapeInfer() const override {
+        return std::make_shared<GenericShapeInfer>();
+    }
+};
+} // namespace
+
 Generic::Generic(const std::shared_ptr<ngraph::Node>& op, const dnnl::engine& eng, WeightsSharing::Ptr &cache)
-    : Node(op, eng, cache, NgraphShapeInferFactory(op, EMPTY_PORT_MASK)), ngraphOp(op) {
+    : Node(op, eng, cache, GenericShapeInferFactory()), ngraphOp(op) {
```

Comment:
Hm, why we cannot just use NgraphShapeInference factory. In case of static node, the shape inference object is not created at all. So when the dynamic shapes support is added, the shape inference will be already here, with no extra code.

Reply:
Every time when we create plugin node, plugin tries to create generic op firstly even if it's not generic. Base `Node` ctor creates `shapeInference` entity using factory. And `NgraphShapeInferFactory` returns `NgraphShapeInfer` object which firstly creates local ngraph op by copying ngraph op with new inputs (`clone_with_new_inputs` method). I mean that each time when we try to create Generic op we always call `clone_with_new_inputs` method for ngraph op.

As you remember `TypeRelaxed` nodes aren't thread safe (Ticket 79554). So if there are operations with quantized bodies (`Loop`, `TensorIterator`, `If`, `Snippets subgraph` in the near future), we always may get segfault for dynamic case. To avoid `NgraphShapeInfer` creation with extra call of `clone_with_new_inputs` I decided to create dummy shape inference entity

Reply:
Exactly. So what's about dynamic `Loop` op with quantized body that contains `TypeRelaxed` nodes? :)